### PR TITLE
Improve gradient test

### PR DESCRIPTION
**Pull request by markdewing**
Allow specifying the finite difference delta from the input file. Motivation for this addition comes from checking the orbital rotation gradients for Bspline SPOs.  The Bspline SPOs can be used with a precision of float or double.  The orbital rotation gradients with float precision were giving a large relative difference, and it turned out the default finite difference delta of 1e-5 was too small.  As the value of delta increased, the relative error decreased, which indicated it was in the numerical noise region.  As an alternate check, the precision of the SPOs was switched to double, and then the relative differences from the default delta were fine.

For improvements in the output formatting, the columns and headers are aligned, as requested in #3793.  The precision of the gradients was increased.  The precision of the relative difference was decreased (pretty much only the exponent matters.).
Making the format of the relative difference more uniform makes it harder to quickly scan for problematic values.  A simple threshold scheme was added that prints !, !!, or !!! after the relative difference to highlight problematic values. The thresholds are set to 10^-3, 10^-2, and 10^-1.  

The documentation now includes sample output.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
desktop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/QMCDrivers/WFOpt/GradientTest.cpp [hunk 1]
```diff
@@ -60,20 +59,62 @@ void GradientTest::run(QMCCostFunctionBase& costFunc, const std::string& root_na
     param_deriv_index_++;
   }
 
-  app_log() << "Param_Name  Value    Numeric            Analytic       Percent" << std::endl;
+  size_t max_name_len = std::string("Param_Name").size();
```
Comment (ye-luo) [inline src/QMCDrivers/WFOpt/GradientTest.cpp:62]: It is better to define a `std::string_view` for "Param_Name" considering it being used twice here and below.

File: src/QMCDrivers/WFOpt/GradientTestInput.h [hunk 2]
```diff
@@ -25,9 +25,11 @@ class GradientTestInput
 
 protected:
   bool do_param_output_ = false;
+  double finite_diff_delta_ = 1e-5;
 
 public:
   bool do_param_output() { return do_param_output_; }
+  double finite_diff_delta() { return finite_diff_delta_; }
```
Comment (ye-luo) [inline src/QMCDrivers/WFOpt/GradientTestInput.h:32]: The overall convention for function naming is starting with a verb. `get_finite_diff_delta` ? also both member functions should be marked const.

**Review by ye-luo — COMMENTED** [review]
Overall looks good to me. Minor change requested.

**Review by PDoakORNL — APPROVED** [review]
LGTM

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**